python/tchannel/handler.py

```diff
@@ -75,6 +77,8 @@ def __init__(self, context, conn):
         self.body = self.message.arg_3
         self.method = self.message.arg_1
         self.connection = conn
+        self.context = context
+        self.id = context.message_id
```

Comment:
You don't need this if you already have the `context`. `id` also makes it look like it belongs to the request.